src/set_theory/ordinal.lean

```diff
@@ -336,18 +336,17 @@ open function
 
 noncomputable lemma embedding_to_cardinal : σ ↪ cardinal.{u} :=
 classical.choice $ embedding.total.resolve_left $ λ ⟨⟨f, hf⟩⟩,
-  let T : Type u := Σ a : σ, quotient.out (@inv_fun _ ⟨0⟩ _ f a) in
-  let ⟨x, hx⟩ := inv_fun_surjective hf (cardinal.mk (set T)) in
-  have hT : set T ≃ (@quotient.out (Type u) _ (@inv_fun _ ⟨0⟩ _ f x)),
-    by simp only [T, hx, cardinal.mk];
-      exact (classical.choice (@quotient.mk_out (Type u) _ _)).symm,
-  have e : set T ↪ T, from ⟨λ s, ⟨x, hT s⟩, λ _, by simp⟩,
-  cantor_injective e.1 e.2
-
-private def well_order_r : σ → σ → Prop :=
+  let g : σ → cardinal.{u} := @inv_fun _ ⟨0⟩ _ f in
```

Comment:
You can avoid the `@` if you put an inhabited instance on `cardinal` (which should exist in any case)